content/guides/kubernetes/app-enhancements-spring-k8s.md

```diff
@@ -0,0 +1,1151 @@
+- Health checks using Application Probes
+  - readinessProbe
+  - livenessProbe
+  - startupProbe
+- Reporting application state using Spring Boot Actuators
+- Service discovery across namespaces using DiscoveryClient
+- Exposing API documentation using Swagger UI
+- Building a Docker image using best practices
+- Layering JARs using the Spring Boot plugin
+- Observing the application using Prometheus exporters
+
+## Reference Architecture
+
+The reference architecture demonstrates an organization where each unit has its
+own application designed using a microservices architecture. Microservices are
+exposed as REST APIs utilizing Spring Boot in an embedded Tomcat server and
+deployed on Kubernetes.
+
+Each microservice is deployed in its own namespace. Placing microservices in
+separate namespaces allows logical grouping that makes it easier to manage
+access privileges by assigning them only to a team responsible for a particular
+application. Spring Cloud Kubernetes Discovery Client makes internal
+communication between microservices seamless. It communicates with the
+Kubernetes API to discover the IPs of all services, running in Pods.
+
+The application implemented in this Reference Architecture is built with several
+open source components, commonly found in most Spring Boot microservices
+deployments. These include:
+
+- [Spring Cloud
+  Kubernetes](https://github.com/spring-cloud/spring-cloud-kubernetes#discoveryclient-for-kubernetes):
+  Provides integration with the Kubernetes API server to allow service discovery,
+  configuration and load balancing used by Spring Cloud.
+
+- [Spring Cloud](https://github.com/spring-cloud): provides tools for developers
+  to quickly build common patterns in distributed systems.
+
+  - [OpenFeign](https://github.com/OpenFeign/feign) - Java to HTTP client binder
+  - Ribbon (via Kubernetes services) - client-side load balancing in calls to
+    another microservice
+  - [Zuul](https://github.com/Netflix/zuul) - routing and filtering requests to
+    a microservice
+  - [Sleuth](https://github.com/spring-cloud/spring-cloud-sleuth) - a
+    distributed tracing tool
+  - [Swagger](https://swagger.io/) - a set of open-source tools built around the
+    OpenAPI Specification that can design, build, document and consume REST APIs
+  - [Swagger UI](https://swagger.io/tools/swagger-ui/) - provides interaction
+    and visualization of API resources without writing custom logic.
+
+![Microservices Landscape](/images/guides/kubernetes/app-enhancements/spring-boot-micro-services-landscape.png)
+
+## Reference Architecture Environment
+
+Each microservice runs in its own container, one container per pod and one pod
+per service replica. The application is built using a microservices architecture and
+represented by replicated containers calling each other.
+
+![Microservices Stack](/images/guides/kubernetes/app-enhancements/spring-boot-micro-services-stack.png)
```

Comment:
Can we crop these images to there isn't the extra whitespace around the border? This isn't the only image but it's one example.